Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -282,6 +282,24 @@ class DatabaseContainer: NSPersistentContainer, @unchecked Sendable {
             }
         }
     }
+    
+    func readAndWait<T>(_ actions: (DatabaseSession) throws -> T) throws -> T {
+        let context = viewContext
+        var result: T?
+        var readError: Error?
+        context.performAndWait {
+            do {
+                result = try actions(context)
+            } catch {
+                readError = error
+            }
+        }
+        if let result {
+            return result
+        } else {
+            throw readError ?? ClientError.Unknown()
+        }
+    }
```

Comment:
Throwing performAndWait is iOS 15 or later, therefore this is not so pretty at the moment.